src/app/register/page.tsx

```diff
@@ -12,9 +19,49 @@ const Registration = () => {
   });
 
   const [errorMessage, setError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
   const router = useRouter();
 
+  const isDisabled = Boolean(emailError || passwordError || passwordError);
+
+  const validateEmail = () => {
```

Comment:
There should be an alert in your head when you copy exactly the same code to another component. This should be abstracted to some component called `EmailInput` in this case. For some other use cases it could be abstracted to utils or to hooks as a custom hook